src/browser/pages/vscode.ts

```diff
@@ -155,14 +203,50 @@ export function setBodyBackgroundToThemeBackgroundColor(document: Document, loca
     )
   }
 
-  document.body.style.background = editorBgColor
+  _document.body.style.background = editorBgColor
 
   return null
 }
 
+/**
+ * A helper function to encapsulate all the
+ * logic used in this file.
+ *
+ * We purposely include all of this in a single function
+ * so that it's easier to test.
+ */
+export function main(_document: Document | undefined, _window: Window | undefined, _localStorage: Storage | undefined) {
+  const errorMsgPrefix = "[vscode]"
+
+  if (!_document) {
+    throw new Error(`${errorMsgPrefix} Could not run code-server vscode logic. document is undefined.`)
+  }
+
+  if (!_window) {
+    throw new Error(`${errorMsgPrefix} Could not run code-server vscode logic. window is undefined.`)
+  }
+
+  if (!_localStorage) {
+    throw new Error(`${errorMsgPrefix} Could not run code-server vscode logic. localStorage is undefined.`)
+  }
+
+  const options = getOptions()
+  const nlsConfig = getNlsConfiguration(_document, options.base)
+
+  const loader = getConfigurationForLoader({
+    nlsConfig,
+    options,
+    _window,
+    origin: _window.location.origin,
```

Comment:
I suppose since we ended up needing `window` we could skip `origin` since it's redundant.

Reply:
Oh yeah true, good catch